pkg/provider/provider.go

```diff
@@ -289,7 +289,7 @@ func (p *Provider) buildServiceAndRoutersForTrafficSplit(t *topology.Topology, c
 
 	logger := p.logger.
 		WithMetadata(tsSvc.Annotations).
```

Comment:
Since you updated the Subject to TrafficSplit, metadata should be updated too, no?